resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/interceptors/MessageSanitizerContainerResponseFilter.java

```diff
@@ -74,5 +76,18 @@ private String escapeXml(String str) {
         return sb.toString();
     }
 
+    private boolean containsHtmlText(ArrayList<Object> list) {
+       for (Iterator<Object> it = list.iterator(); it.hasNext();  ) {
+          Object o = it.next();
+          if (o instanceof String) {
+             String mediaType = (String) o;
+             String[] parts = mediaType.split("/");
+             if (parts.length >= 2 && parts[0].trim().equals("text") && parts[1].trim().startsWith("html")) {
```

Comment:
Changed to

          if (o instanceof String) {
             String mediaType = (String) o;
             String[] partsType = mediaType.split("/");
             if (partsType.length >= 2) {
                String[] partsSubtype = partsType[1].split(";");
                if (partsType[0].trim().equalsIgnoreCase("text") && 
                      partsSubtype[0].trim().toLowerCase().equals("html")) {
                   return true;
                }
             }
          }